target/linux/ramips/dts/APAC1200.dts

```diff
@@ -0,0 +1,178 @@
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *  HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/dts-v1/;
+
+#include "mt7620a.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "v7,apac1200", "ralink,mt7620a-soc";
+	model = "V7 APAC1200";
+
+	chosen {
+		bootargs = "console=ttyS0,57600";
+	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		wlan2g {
+			label = "apac1200:blue:wlan2g";
+			gpios = <&gpio0 1 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan5g {
+			label = "apac1200:yellow:wlan5g";
+			gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partition@0 {
+			label = "u-boot";
+			reg = <0x0 0x30000>;
+			read-only;
+		};
+
+		config: partition@30000 {
+			label = "config";
+			reg = <0x30000 0x10000>;
+			read-only;
+		};
+
+		art: partition@48000 {
+			label = "art";
+			reg = <0x48000 0x8000>;
+			read-only;
+		};
+
+		factory: partition@40000 {
+			label = "factory";
+			reg = <0x40000 0x10000>;
+			read-only;
+		};
+
+		partition@50000 {
+			label = "firmware";
+			reg = <0x50000 0x6ca000>;
+		};
+
+		partition@71a000 {
+			label = "manufacture";
+			reg = <0x71a000 0x96000>;
+		};
+
+		partition@7b0000 {
+			label = "backup";
+			reg = <0x7b0000 0x10000>;
+		};
+
+		partition@7c0000 {
+			label = "storage";
+			reg = <0x7c0000 0x40000>;
+		};
+
+	};
+};
+
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			ralink,group = "i2c", "uartf";
+			ralink,function = "gpio";
+		};
+	};
+};
+
+&ethernet {
+	status = "okay";
+	mtd-mac-address = <&factory 0x4>;
+	mtd-mac-address-increment = <1>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&rgmii1_pins &rgmii2_pins &mdio_pins>;
```

Comment:
One of the rgmii pins groups here is superfluous, since you're only using one rgmii out of two. But I've no idea which one is used. You would need to set the rgmii[1|2] group to function gpio to see which one is related/required.

Reply:
I think I understand what you're asking here, but when I set either of the rgmii groups to function gpio, the switch fails to load:

'''
&pinctrl {
        state_default: pinctrl0 {
                gpio {
                        ralink,group = "i2c", "nd_sd", "rgmii1", "uartf";
                        ralink,function = "gpio";
                };
        };
};
'''
results in:
'''
[    1.840523] rt2880-pinmux pinctrl: pin io24 already requested by pinctrl; cannot claim for 10100000.ethernet
[    1.860171] rt2880-pinmux pinctrl: pin-24 (10100000.ethernet) status -22
[    1.873537] rt2880-pinmux pinctrl: could not request pin 24 (io24) from group rgmii1  on device rt2880-pinmux
[    1.893300] mtk_soc_eth 10100000.ethernet: Error applying setting, reverse things back
[    1.909095] mtk_soc_eth: probe of 10100000.ethernet failed with error -22
'''
and
'''
&pinctrl {
        state_default: pinctrl0 {
                gpio {
                        ralink,group = "i2c", "nd_sd", "rgmii1", "uartf";
                        ralink,function = "gpio";
                };
        };
};
'''
results in
'''
[    1.840542] rt2880-pinmux pinctrl: pin io60 already requested by pinctrl; cannot claim for 10100000.ethernet
[    1.860188] rt2880-pinmux pinctrl: pin-60 (10100000.ethernet) status -22
[    1.873553] rt2880-pinmux pinctrl: could not request pin 60 (io60) from group rgmii2  on device rt2880-pinmux
[    1.893315] mtk_soc_eth 10100000.ethernet: Error applying setting, reverse things back
[    1.909122] mtk_soc_eth: probe of 10100000.ethernet failed with error -22
'''

Reply:
You need to drop the pingroup set to GPIO from `pinctrl-0 = <&rgmii1_pins &rgmii2_pins &mdio_pins>;` as well.